ext/spl/spl_iterators.c

```diff
@@ -1880,6 +1880,9 @@ PHP_METHOD(RegexIterator, accept)
 			}
 
 			result = php_pcre_replace_impl(intern->u.regex.pce, subject, ZSTR_VAL(subject), ZSTR_LEN(subject), replacement_str, -1, &count);
+			if (!replacement_str) {
+				RETURN_FALSE;
```

Comment:
Shouldn't this be a ``RETURN_NULL`` same as on line 1879 (the return should default to returning ``null``) instead of returning ``false``.